### PR TITLE
Carry PR 12008 for contributor

### LINKED ISSUE #8040: docker run -t causes stderr messages to be sent to stdout.
Bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1121689 

Description of problem:
The handling of stdout/stderr stream is not correct in tty mode (works fine in non-tty mode). You can see it on the example bellow. (`ls non_existing_file` prints into stderr stream)

Version-Release number of selected component (if applicable):
docker-1.1.1-1.el7.x86_64

How reproducible:
Always

Steps to Reproduce:
TTY:
[root@localhost ~]# docker run -t -a stderr fedora ls non_existing_file
[root@localhost ~]# docker run -t -a stdout fedora ls non_existing_file
ls: cannot access non_existing_file: No such file or directory
NON-TTY:
[root@localhost ~]# docker run -a stderr fedora ls non_existing_file
ls: cannot access non_existing_file: No such file or directory
[root@localhost ~]# docker run -a stdout fedora ls non_existing_file

Actual results:
In tty mode the output is visible only when STDOUT is attached, in non-tty mode it's visible in STDERR

Expected results:
In booth modes the output should be in STDERR stream.

Additional info:
When I use `ls /` it prints the info correctly into STDOUT in booth modes.

### PR DESCRIPTION
**Pull request by moxiegirl**
Closes #8040 
Carry PR #12008 

Signed-off-by: Mary Anthony mary@docker.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
LGTM

File: docs/sources/reference/commandline/cli.md [hunk 1]
```diff
@@ -84,14 +84,20 @@ be set to the non-default value by explicitly setting them to `false`:
...
+    $ docker run -a stdin -a stdout -a stderr ubuntu /bin/ls
 
-Sometimes this can use a more complex value string, as for `-v`:
+Sometimes, multiple options can call for a more complex value string as for `-v`:
 
     $ docker run -v /host:/container example/mysql
 
+> **Note**:
+> Do not use the `-t` and `-a stderr` options together due to limitations
+> in the `pty` implementation. All `stderr` in `pty` mode simply goes to `stdout``.
```
Comment (thaJeztah) [inline docs/sources/reference/commandline/cli.md]: Extraneous ``` after stdout (probably not problematic)

Comment (jfrazelle) [inline docs/sources/reference/commandline/cli.md]: damn nice find

Comment (moxiegirl) [inline docs/sources/reference/commandline/cli.md]: The man has eyes like a hawk.

Comment (thaJeztah) [inline docs/sources/reference/commandline/cli.md]: Nah, just lucky spotting it. Sometimes feel bad putting y'all to work to fix them 😕

**Comment by moxiegirl** [comment]
LGTM

**Pull request merged**